api/src/main/java/io/druid/data/input/impl/StringInputRowParser.java

```diff
@@ -57,6 +57,7 @@ public StringInputRowParser(
     this.parseSpec = parseSpec;
     this.mapParser = new MapInputRowParser(parseSpec);
     this.parser = parseSpec.makeParser();
+    parser.startFileFromBeginning();
```

Comment:
StringInputRowParser is used by parser options that aren't file-oriented (you can use it on streams etc) so this isn't a good place to put this. imo, this should replace `reset()` and be called in places that `reset()` is currently called (like FileIteratingFirehose). With one addition: it needs to be called before the very first file too.